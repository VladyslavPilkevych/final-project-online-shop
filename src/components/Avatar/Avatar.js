import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { NavLink } from 'react-router-dom';

import { ReactComponent as SignInIcon } from '../../assets/icons/Sign-in.svg';
import AvatarIcon from '../../assets/avatar/avatar.png';
import { removeUserDataAC } from '../../store/actionCreators/userAC';

import styles from './Avatar.module.scss';

function Avatar() {
  const user = useSelector((state) => state.user.user);
  const dispatch = useDispatch();

  return (
    <div>
      {!user ? (
        <NavLink to="/sign-in">
          <SignInIcon className={styles.signInIcon} />
        </NavLink>
      ) : (
        // eslint-disable-next-line jsx-a11y/no-noninteractive-element-to-interactive-role
        user && <img src={AvatarIcon} alt="AvatarIcon" role="button" tabIndex="0" onClick={() => dispatch(removeUserDataAC())} />
      )}
    </div>
  );
}

export default Avatar;
