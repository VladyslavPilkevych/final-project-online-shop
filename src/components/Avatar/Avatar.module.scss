@import '../../assets/styles/global.module.scss';

.signInIcon {
  stroke: $black;
}
@media (max-width: 768px) and (min-width: 376px) {
  .signInIcon {
    stroke: $white;
  }
}
@media (max-width: 375px) and (min-width: 240px) {
  .signInIcon {
    stroke: $white;
  }
}
