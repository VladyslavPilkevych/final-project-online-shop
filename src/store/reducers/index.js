import { combineReducers } from 'redux';
import preloaderReducer from './preloaderReducer';
import productsReducer from './productsReducer';
import menuReducer from './menuReducer';
import userReducer from './userReducer';
import searchReducer from './searchReducer';
import cartReducer from './cartReducer';
import miniMenuReducer from './miniMenuReducer';
import filtersCategoriesReducer from './filtersCategoriesReducer';

const reducer = combineReducers({
  loader: preloaderReducer,
  products: productsReducer,
  menu: menuReducer,
  miniMenu: miniMenuReducer,
  filtersCategories: filtersCategoriesReducer,
  user: userReducer,
  search: searchReducer,
  cart: cartReducer,
});

export default reducer;
