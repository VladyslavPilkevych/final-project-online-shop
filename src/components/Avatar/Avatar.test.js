/* eslint-disable react/react-in-jsx-scope */
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../store';
import Avatar from './Avatar';

function Component() {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <Avatar />
      </Provider>
    </BrowserRouter>
  );
}

describe('Snapshot avatar should render', () => {
  test('Snapshot avatar is rendered', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});
