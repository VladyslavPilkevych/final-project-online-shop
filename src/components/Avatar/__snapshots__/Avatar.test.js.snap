// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot avatar should render Snapshot avatar is rendered 1`] = `
<DocumentFragment>
  <div>
    <a
      class=""
      href="/sign-in"
    >
      <svg
        class="signInIcon"
      >
        Sign-in.svg
      </svg>
    </a>
  </div>
</DocumentFragment>
`;
